app/controllers/application_controller.rb

```diff
@@ -4,7 +4,13 @@ class ApplicationController < ActionController::Base
   private
     def set_current_user
       if session[:current_user]
-        @current_user = User.find(session[:current_user])
+        begin
+          @current_user = User.find(session[:current_user])
+        rescue
+          session[:current_user] = nil
+          @current_user = nil
+          redirect_to users_path, notice: "You have logged off."
```

Comment:
Regarding user experience, it would be a bit strange to see this message without clicking "Log Off", because to "log off" is action done by the user. This message might be less confusing:

```suggestion
          redirect_to users_path, notice: "Authentication has expired; you are no longer logged in."
```